src/devices/Mcp25xxx/Mcp25xxx.cs

```diff
@@ -0,0 +1,404 @@
+            }
+        }
+
+        /// <summary>
+        /// Reads the current value of Rx0BF pin.
+        /// </summary>
+        public PinValue Rx0BfPin
+        {
+            get
+            {
+                VerifyPinConfigured(_rx0bf, nameof(Rx0BfPin));
+                return _masterGpioController.Read(_rx0bf);
+            }
+        }
+
+        /// <summary>
+        /// Reads the current value of Rx1BF pin.
+        /// </summary>
+        public PinValue Rx1BfPin
+        {
+            get
+            {
+                VerifyPinConfigured(_rx1bf, nameof(Rx1BfPin));
+                return _masterGpioController.Read(_rx1bf);
+            }
+        }
+
+        /// <summary>
+        /// Resets the internal registers to the default state and sets Configuration mode.
+        /// </summary>
+        public void Reset()
+        {
+            const byte instructionFormat = 0xC0;
+            _spiDevice.WriteByte(instructionFormat);
+        }
+
+        /// <summary>
+        /// Reads data from the register beginning at the selected address.
+        /// </summary>
+        /// <param name="address">The address to read.</param>
+        /// <returns>The value of address read.</returns>
+        public byte Read(Address address)
+        {
+            const byte instructionFormat = 0x03;
+            const byte dontCare = 0x00;
+            ReadOnlySpan<byte> writeBuffer = stackalloc byte[] { instructionFormat, (byte)address, dontCare };
+            Span<byte> readBuffer = stackalloc byte[3];
+            _spiDevice.TransferFullDuplex(writeBuffer, readBuffer);
+            return readBuffer[2];
+        }
+
+        /// <summary>
+        /// When reading a receive buffer, reduces the overhead of a normal READ
+        /// command by placing the Address Pointer at one of four locations for the receive buffer.
+        /// </summary>
+        /// <param name="addressPointer">The Address Pointer to one of four locations for the receive buffer.</param>
+        /// <param name="byteCount">Number of bytes to read.  This must be one or more to read.</param>
+        /// <returns>The value of address read.</returns>
+        public byte[] ReadRxBuffer(RxBufferAddressPointer addressPointer, int byteCount = 1)
+        {
+            if (byteCount < 1)
+            {
+                throw new ArgumentException($"Invalid number of bytes {byteCount}.", nameof(byteCount));
+            }
+
+            const int StackThreshold = 32;  // Usually won't read more than this at a time.
+
+            Span<byte> writeBuffer = byteCount < StackThreshold ?
+               stackalloc byte[byteCount + 1] :
+               new byte[byteCount + 1];
+
+            Span<byte> readBuffer = byteCount < StackThreshold ?
+               stackalloc byte[byteCount + 1] :
+               new byte[byteCount + 1];
+
+            byte instructionFormat = (byte)(0x90 | ((byte)addressPointer << 1));
```

Comment:
what's `0x90`? usually `0x80` means read

Reply:
See Table 5-1 SPI Instruction Set in datasheet.  Command **Read Rx  Buffer** has an Instruction Format base value of 0x90 and the 2nd/3rd bits are used to point to an address for reading.  I'll update with a variable to make more sense.